packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/ObjectIdImpl.kt

```diff
@@ -4,65 +4,40 @@ import io.realm.kotlin.internal.interop.OBJECT_ID_BYTES_SIZE
 import io.realm.kotlin.internal.platform.epochInSeconds
 import io.realm.kotlin.internal.util.HEX_PATTERN
 import io.realm.kotlin.internal.util.parseHex
-import io.realm.kotlin.internal.util.toHexString
 import io.realm.kotlin.types.ObjectId
 import io.realm.kotlin.types.RealmInstant
-import kotlinx.atomicfu.AtomicInt
-import kotlinx.atomicfu.atomic
-import kotlin.random.Random
+import kotlin.time.Duration.Companion.nanoseconds
+import kotlin.time.Duration.Companion.seconds
+
+private fun RealmInstant.toEpochMillis(): Long = (epochSeconds.seconds + nanosecondsOfSecond.nanoseconds).inWholeMilliseconds
 
 @Suppress("MagicNumber")
```

Comment:
Seems like you have also eliminated most of the `MagicNumber`s. Don't know how many are left, but consider moving the `@Suppress` to the actual remaining places.